presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/README.md

```diff
@@ -0,0 +1,148 @@
+# Presto planner and optimizer tests
+
+Testing planner and optimizer functionality generally requires generating a
+plan and then validating it against a set of expectations. The challenge is
+expressing the expectations in a way that's concise and resilient to changes in
+the objects Presto uses to represent the plan and the code that generates the
+plan.
+
+Unfortunately, that list includes the following:
+
+* The parser
+* The analyzer
+* The planner
+* Some or all of the optimizers
+* The plan objects themselves
+
+The plan testing framework attempts to solve the problem of specifiying
+expectations about query plans in tests.
+
```

Comment:
Not sure I understand what the plan testing framework is for. How are query plans in tests related to the plan testing framework. I would need a better understanding before suggesting an edit. Could you please send me a little more info?